engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Type.java

```diff
@@ -118,6 +118,13 @@ public boolean isBuiltin() {
   }
 
   public Type getSupertype() {
+    if (supertype == null) {
+      if (builtin) {
+        return null;
+      }
+      var ctx = EnsoContext.get(null);
```

Comment:
Calling `get(null)` might affect a performance a bit. Running [benchmarks to verify](https://github.com/enso-org/enso/actions/runs/5546262974).

Reply:
Benchmarks seem to be unaffected by this change. Thanks @Akirathan for the new benchmarking tool.

![Sample benchmark comparition](https://github.com/enso-org/enso/assets/26887752/41836cf9-500d-4c81-bf57-d5595dad0fa8)